hadoop-ozone/common/src/test/java/org/apache/hadoop/ozone/om/helpers/TestOMNodeDetails.java

```diff
@@ -0,0 +1,421 @@
+    assertEquals(HOST_ADDRESS + ":9874", urlWithoutFlush.getAuthority());
+    assertNotNull(urlWithoutFlush.getQuery());
+    assertTrue(urlWithoutFlush.getQuery().contains("flushBeforeCheckpoint=false"));
+
+    URL urlWithFlush = nodeDetails.getOMDBCheckpointEndpointUrl(true, true);
+    assertNotNull(urlWithFlush);
+    assertTrue(urlWithFlush.getQuery().contains("flushBeforeCheckpoint=true"));
+    assertTrue(urlWithFlush.getQuery().contains("includeSnapshotData=true"));
+  }
+
+  /**
+   * Test getOMDBCheckpointEndpointUrl for HTTPS.
+   */
+  @Test
+  public void testGetOMDBCheckpointEndpointUrlHttps() throws IOException {
+    OMNodeDetails nodeDetails = new OMNodeDetails.Builder()
+        .setOMServiceId(OM_SERVICE_ID)
+        .setOMNodeId(OM_NODE_ID)
+        .setHostAddress(HOST_ADDRESS)
+        .setRpcPort(RPC_PORT)
+        .setRatisPort(RATIS_PORT)
+        .setHttpAddress(HOST_ADDRESS + ":9874")
+        .setHttpsAddress(HOST_ADDRESS + ":9875")
+        .build();
+
+    URL url = nodeDetails.getOMDBCheckpointEndpointUrl(false, false);
+    assertNotNull(url);
+    assertEquals("https", url.getProtocol());
+    assertEquals(HOST_ADDRESS + ":9875", url.getAuthority());
+  }
+
+  /**
+   * Test getOMNodeAddressFromConf.
+   */
+  @Test
+  public void testGetOMNodeAddressFromConf() {
+    OzoneConfiguration conf = new OzoneConfiguration();
+    
+    String configKey = "ozone.om.address.om-service.om-01";
+    String expectedAddress = "localhost:9862";
+    conf.set(configKey, expectedAddress);
+
+    String address = OMNodeDetails.getOMNodeAddressFromConf(conf, "om-service", "om-01");
+    assertEquals(expectedAddress, address);
+
+    String missingAddress = OMNodeDetails.getOMNodeAddressFromConf(conf, "nonexistent", "node");
+    assertNull(missingAddress);
+  }
+
+  /**
+   * Test getOMNodeDetailsFromConf with valid configuration.
+   */
+  @Test
+  public void testGetOMNodeDetailsFromConfValid() throws Exception {
+    OzoneConfiguration conf = new OzoneConfiguration();
+    
+    String serviceId = "om-service";
+    String nodeId = "om-01";
+    
+    conf.set(ConfUtils.addKeySuffixes(OMConfigKeys.OZONE_OM_ADDRESS_KEY, serviceId, nodeId), 
+        "localhost:9862");
+    conf.set(ConfUtils.addKeySuffixes(OMConfigKeys.OZONE_OM_RATIS_PORT_KEY, serviceId, nodeId), 
+        "9873");
+    conf.set(ConfUtils.addKeySuffixes(OMConfigKeys.OZONE_OM_HTTP_ADDRESS_KEY, serviceId, nodeId),
+        "localhost:9874");
+    conf.set(ConfUtils.addKeySuffixes(OMConfigKeys.OZONE_OM_HTTPS_ADDRESS_KEY, serviceId, nodeId),
+        "localhost:9875");
+
+    OMNodeDetails nodeDetails = OMNodeDetails.getOMNodeDetailsFromConf(conf, serviceId, nodeId);
+
+    assertNotNull(nodeDetails);
+    assertEquals(serviceId, nodeDetails.getServiceId());
+    assertEquals(nodeId, nodeDetails.getNodeId());
+    assertEquals(9862, nodeDetails.getRpcPort());
+    assertEquals(9873, nodeDetails.getRatisPort());
+  }
```

Comment:
Can you add two negative tests here

```suggestion
    nodeDetails = OMNodeDetails.getOMNodeDetailsFromConf(conf, serviceId, null);
    assertNull(nodeDetails);
    nodeDetails = OMNodeDetails.getOMNodeDetailsFromConf(conf, null, nodeId);
    assertNull(nodeDetails);
  }
```

Reply:
Thank you for your suggestion. I think we could put this two negative test to testGetOMNodeDetailsFormConfMissing(). Is this a good way?